File: spring-ai-alibaba-data-agent-management/src/main/java/com/alibaba/cloud/ai/dataagent/entity/ModelConfig.java
/*
 * Copyright 2024-2025 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.alibaba.cloud.ai.dataagent.entity;

import com.alibaba.cloud.ai.dataagent.enums.ModelType;
import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class ModelConfig {

	private Integer id;

	// 厂商标识 (方便前端展示回显，实际调用主要靠 baseUrl)
	private String provider;

	// 关键配置
	private String baseUrl;

	private String apiKey;

	private String modelName;

	private Double temperature;

	private Boolean isActive = false;

	private Integer maxTokens;

	// 模型类型
	// 可选值："CHAT", "EMBEDDING"
	@Column(nullable = false)
	private ModelType modelType;

	@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
	private LocalDateTime createdTime;

	@JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "GMT+8")
	private LocalDateTime updatedTime;

	// 0=未删除, 1=已删除
	private Integer isDeleted;

}

File: spring-ai-alibaba-data-agent-frontend/src/services/modelConfig.ts
/*
 * Copyright 2024-2025 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import axios from 'axios';
import type { ApiResponse } from './common';

export interface ModelConfig {
  id?: number;
  provider: string; // e.g. "openai", "deepseek"
  apiKey: string;
  baseUrl: string;
  modelName: string;
  modelType: string; // "CHAT" or "EMBEDDING"
  temperature?: number;
  maxTokens?: number;
  isActive?: boolean;
}

const API_BASE_URL = '/api/model-config';

class ModelConfigService {
  /**
   * 获取模型配置列表
   */
  async list(): Promise<ModelConfig[]> {
    const response = await axios.get<ApiResponse<ModelConfig[]>>(`${API_BASE_URL}/list`);
    return response.data.data || [];
  }

  /**
   * 新增模型配置
   * @param config 模型配置对象
   */
  async add(config: Omit<ModelConfig, 'id'>): Promise<boolean> {
    console.log('config: ' + config);
    const response = await axios.post<ApiResponse<string>>(`${API_BASE_URL}/add`, config);
    return response.data.success;
  }

  /**
   * 更新模型配置
   * @param config 模型配置对象
   */
  async update(config: ModelConfig): Promise<boolean> {
    const response = await axios.put<ApiResponse<string>>(`${API_BASE_URL}/update`, config);
    return response.data.success;
  }

  /**
   * 删除模型配置
   * @param id 配置ID
   */
  async delete(id: number): Promise<ApiResponse<string>> {
    const response = await axios.delete<ApiResponse<string>>(`${API_BASE_URL}/${id}`);
    return response.data;
  }

  /**
   * 启用/切换模型配置
   * @param id 配置ID
   */
  async activate(id: number): Promise<boolean> {
    const response = await axios.post<ApiResponse<string>>(`${API_BASE_URL}/activate/${id}`);
    return response.data.success;
  }
}

export default new ModelConfigService();
